angular/projects/global/src/lib/components/table-template/table-template-utils.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from '@angular/core';
+import { Router, Params } from '@angular/router';
+import { PlatformLocation } from '@angular/common';
+import {
+  TableObject,
+  DEFAULT_TABLE_PAGE_SIZE,
+  DEFAULT_TABLE_CURRENT_PAGE,
+  DEFAULT_TABLE_SORT_BY,
+  DEFAULT_TABLE_KEYWORDS
+} from './table-object';
+
+@Injectable()
+export class TableTemplateUtils {
+  constructor(private platformLocation: PlatformLocation, private router: Router) {}
+
+  public updateUrl(tableObject: TableObject) {
+    let currentUrl = this.router.url;
+    currentUrl = (this.platformLocation as any).getBaseHrefFromDOM() + currentUrl.slice(1);
+    currentUrl = currentUrl.split(';')[0];
+    currentUrl += `;currentPage=${tableObject.currentPage};pageSize=${tableObject.pageSize}`;
+    if (tableObject.keywords !== '') {
+      currentUrl += `;keywords=${tableObject.keywords}`;
+    }
+    if (tableObject.sortBy !== '' && tableObject.sortBy !== null) {
+      currentUrl += `;sortBy=${tableObject.sortBy}`;
+    }
+    if (tableObject.filter !== null && tableObject.filter !== {}) {
+      Object.keys(tableObject.filter).forEach(key => {
+        if (tableObject.filter[key] === true || tableObject.filter[key] === false) {
+          currentUrl += `;${key}=${tableObject.filter[key]}`;
+        } else {
+          currentUrl += `;${key}=`;
+          tableObject.filter[key].split(',').forEach(item => {
+            currentUrl += `${item},`;
+          });
+          currentUrl = currentUrl.slice(0, -1);
+        }
+      });
+    }
+    currentUrl += ';ms=' + new Date().getTime();
```

Comment:
I've seen this pattern before.  As far as I know it shouldn't be necessary? As long as we subscribe to the route and update the page accordingly when params change.